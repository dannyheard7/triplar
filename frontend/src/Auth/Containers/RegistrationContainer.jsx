import React from "react";

import RegistrationForm from "Auth/Components/RegistrationForm";
import api from "Auth/utils/auth.api";
import FormContainer from "Forms/Containers/FormContainer";
import {Helmet} from "react-helmet";

export default class RegistrationContainer extends React.Component {

    constructor(props) {
        super(props);

        this.onSuccess = this.onSuccess.bind(this);
    }

    onSuccess(data) {
        this.props.history.push('/login')
    }

    render() {
        return (
            <div>
                <Helmet>
                    <title>Register | Triplar</title>
                </Helmet>
                <FormContainer onSuccess={this.onSuccess} apiFunction={api.registerUser}>
                    <RegistrationForm />
                </FormContainer>
            </div>
        );
    }
}
