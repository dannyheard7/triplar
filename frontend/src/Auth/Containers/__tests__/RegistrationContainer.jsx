import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import RegistrationContainer from "../RegistrationContainer";

const renderer = new ShallowRenderer();

describe('<RegistrationContainer />', () => {
    const props = {
        errors: null
    };

    test('renders correctly', () => {
        const result = renderer.render(<RegistrationContainer {...props} />);
        expect(result).toMatchSnapshot();
    });
});