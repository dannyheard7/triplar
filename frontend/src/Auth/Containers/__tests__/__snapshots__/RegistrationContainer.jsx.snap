// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationContainer /> renders correctly 1`] = `
<Route>
  [Function]
</Route>
`;
